ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -0,0 +1,210 @@
+
+Users can request permissions in two ways - 1) By performing a read operation like read or readText using Async clipboard API. 2) By explicitly requesting "clipboard-read" permission, the API for this is still under discussion (https://github.com/w3c/permissions/issues/158)
+
+##### Pros
+1. More defensive approach, guards against potential misuse of clipboard change event.
+2. Ensures user consent before accessing clipboard data.
+
+##### Cons
+1. Not a clear user flow for requesting permissions.
+2. Additional complexity for both implementation and web authors.
+
+To simply further, we can prompt user for permissions as soon as the "addEventListener" method is called with "clipboardchange" in case the permissions are not already granted. This is still open for discussion as it is not a common pattern to prompt user for permissions when attaching event listeners.
+
+#### 5.1.2 Approach 2 - No permission required
+Since no data is being sent as part of the clipboardchange event, it can be argued that we don't need any permission to simply know when clipboard contents change. This will simplify the user flow as they don't need to explicitly ask for permissions before listening to the event.
+
+##### Pros
+1. Simpler implementation and user experience
+
+##### Cons
+1. Open to privacy attacks which simply monitor clipboard changes without user consent.
+
+#### 5.1.3 Conclusion
+We favour Approach 1 i.e. having clipboard-read permission required to listen to clipboardchange event, because it has more provisions which safegaurds user privacy.
+
+### 5.2 Page focus requirement
+As per the [current spec](https://www.w3.org/TR/clipboard-apis/#clipboard-event-clipboardchange), we should not fire "clipboardchange" event when a page is not is focus. This is in-line with the current behavior where async clipboard API is not accessible unless the given page is in focus. We do fire "clipboardchange" event when the page regains focus, incase the clipboard contents had changed when the page was out of focus. Note that even if the clipboard had changed multiple times while the page was out of focus, we will only fire a single "clipboardchange" event when the page regains focus. This is because the event is designed to indicate that the clipboard contents are different from what they were when the page lost focus, rather than tracking every individual change that occurred while the page was out of focus.
+
+#### 5.2.1 Approach 1 - Page required to be in focus to receive event
+
+##### Pros
+1. More defensive and prevents background scripts from misuse of clipboardchange events.\
+1.a) Example - Malicious scripts can detect when a cryptocurrency wallet address is copied to clipboard and replace it with attacker's address and that way redirect payment to attacker's account.  
```

Comment:
It would be great if we can use an example from the use cases we listed previously.

Reply:
I can't think of malicious use of the clipboardchange from the listed use cases - remote desktop clipboard sync and rich web editors. Maybe in case of remote desktop sync, a bad actor running natively on the remote machine can misuse clipboard change notifications triggering in the background of client's local machine, however in that case the remote desktop webapp can prevent sending clipboard updates to the remote machine when page is out of focus.